### PR TITLE
refactor: messagingpush module test classes

### PR DESCRIPTION
**Pull request by mrehan27**
part of [MBL-392](https://linear.app/customerio/issue/MBL-392/decouple-push-tests-from-tracking-module)

### Changes

- Removed test dependencies from module's gradle file as they are now included in the common gradle file for test dependencies
- Updated tests to match package of their respective classes
- Updated tests to work with `common-test` updates

### Test Status

All tests for `messagingpush` module are passing ✅

### Notes

See [parent PR](https://github.com/customerio/customerio-android/pull/373) for more details and context.

### CODE REVIEW

File: messagingpush/src/androidTest/java/io/customer/messagingpush/activity/NotificationClickReceiverActivityTest.kt [hunk 1]
```diff
@@ -5,29 +5,41 @@ import android.os.Bundle
...
+                    sdk {
+                        overrideDependency<EventBus>(mockk(relaxed = true))
+                        overrideDependency<PushMessageProcessor>(mockk(relaxed = true))
+                    }
+                }
+            }
+        )
 
-        di.overrideDependency(PushMessageProcessor::class.java, pushMessageProcessorMock)
+        eventBus = SDKComponent.eventBus
```
Comment (Shahroz16) [inline messagingpush/src/androidTest/java/io/customer/messagingpush/activity/NotificationClickReceiverActivityTest.kt:41]: why do we need to mock this?

File: messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt [hunk 2]
```diff
@@ -1,103 +1,79 @@
...
-        super.teardown()
-    }
-
     @Test
     fun initialize_givenNoConfig_expectDefaultValues() {
-        module = ModuleMessagingPushFCM()
+        ModuleMessagingPushFCM().attachToSDKComponent()
 
-        SDKComponent.modules[ModuleMessagingPushFCM.MODULE_NAME] = module
+        val moduleConfig = SDKComponent.moduleConfig
```
Comment (Shahroz16) [inline messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt:43]: cant we just shorten it to following

```
val moduleConfig = ModuleMessagingPushFCM().attachToSDKComponent()
```

File: messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt [hunk 3]
```diff
@@ -1,103 +1,79 @@
...
     @Test
     fun initialize_givenEmptyConfig_expectDefaultValues() {
-        module = ModuleMessagingPushFCM(
+        ModuleMessagingPushFCM(
             moduleConfig = MessagingPushModuleConfig.default()
-        )
+        ).attachToSDKComponent()
 
-        SDKComponent.modules[ModuleMessagingPushFCM.MODULE_NAME] = module
+        val moduleConfig = SDKComponent.moduleConfig
```
Comment (Shahroz16) [inline messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt:56]: same as above

**Review by Shahroz16 — APPROVED** [review]
Minor improvement

File: messagingpush/src/androidTest/java/io/customer/messagingpush/activity/NotificationClickReceiverActivityTest.kt (hunk 1 above)
Comment (mrehan27) [reply messagingpush/src/androidTest/java/io/customer/messagingpush/activity/NotificationClickReceiverActivityTest.kt]: Removed

File: messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt (hunk 2 above)
Comment (mrehan27) [reply messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt]: Updated

File: messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt (hunk 3 above)
Comment (mrehan27) [reply messagingpush/src/test/java/io/customer/messagingpush/ModuleMessagingConfigTest.kt]: Updated

**Pull request merged**